packages/env/lib/config/get-config-from-environment-vars.js

```diff
@@ -84,3 +77,33 @@ function getPortFromEnvironmentVariable( varName ) {
 
 	return port;
 }
+
+/**
+ * Parses the lifecycle script environment variables.
+ *
+ * @return {Object.<string, string>} The parsed lifecycle scripts.
+ */
+function getLifecycleScriptOverrides() {
+	const lifecycleScripts = {};
+
+	// Find all of the lifecycle script overrides and parse them.
+	for ( const env in process.env ) {
+		const match = env.match( /WP_ENV_LIFECYCLE_SCRIPT_([A-Za-z0-9_]+)/ );
```

Comment:
I'd prefer using a manual map of allowed lifecycle scripts -- I think it's a bit clearer and safer.

Like this:

```js
const lifecycleHooks = {
  WP_ENV_LIFECYCLE_SCRIPT_AFTER_START: 'afterStart'
}
```

Reply:
In working on the config validation I agree that it would make more sense to use a map.